src/NovelRT.Interop/Timing/NrtStepTimer.cpp

```diff
@@ -3,150 +3,138 @@
+uint32_t Nrt_StepTimer_getFramesPerSecond(NrtStepTimerHandle timer)
+{
+    auto* time = reinterpret_cast<NovelRT::Timing::StepTimer*>(timer);
+    return time->getFramesPerSecond();
+}
+
+NrtBool Nrt_StepTimer_getIsFixedTimeStep(NrtStepTimerHandle timer)
+{
+    auto* time = reinterpret_cast<NovelRT::Timing::StepTimer*>(timer);
+    return time->isFixedTimeStep() ? NRT_TRUE : NRT_FALSE;
+}
+
+NrtResult Nrt_StepTimer_setIsFixedTimeStep(NrtStepTimerHandle timer, int32_t input)
+{
+    if (timer == nullptr)
     {
-        if (timer == nullptr)
-        {
-            Nrt_setErrIsNullInstanceProvidedInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        NovelRT::Timing::StepTimer time = reinterpret_cast<NovelRT::Timing::StepTimer&>(timer);
-        time.isFixedTimeStep() = input;
-        return NRT_SUCCESS;
+        Nrt_setErrIsNullInstanceProvidedInternal();
+        return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
     }
 
-    NrtResult Nrt_StepTimer_resetElapsedTime(NrtStepTimerHandle timer)
+    auto* time = reinterpret_cast<NovelRT::Timing::StepTimer*>(timer);
+    time->isFixedTimeStep() = input;
+    return NRT_SUCCESS;
+}
+
+NrtResult Nrt_StepTimer_resetElapsedTime(NrtStepTimerHandle timer)
+{
+    if (timer == nullptr)
     {
-        if (timer == nullptr)
-        {
-            Nrt_setErrIsNullInstanceProvidedInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        NovelRT::Timing::StepTimer time = reinterpret_cast<NovelRT::Timing::StepTimer&>(timer);
-        time.resetElapsedTime();
-        return NRT_SUCCESS;
+        Nrt_setErrIsNullInstanceProvidedInternal();
+        return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
     }
 
-    NrtResult Nrt_StepTimer_tick(NrtStepTimerHandle timer, NrtUtilitiesEventWithTimestampHandle event)
+    auto* time = reinterpret_cast<NovelRT::Timing::StepTimer*>(timer);
+    time->resetElapsedTime();
+    return NRT_SUCCESS;
+}
+
+NrtResult Nrt_StepTimer_tick(NrtStepTimerHandle timer, NrtUtilitiesEventWithTimestampHandle event)
+{
+    if (event == nullptr)
     {
-        if (event == nullptr)
-        {
-            Nrt_setErrIsNullInstanceProvidedInternal();
-            return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
-        }
-
-        NovelRT::Timing::StepTimer time = reinterpret_cast<NovelRT::Timing::StepTimer&>(timer);
-        NovelRT::Utilities::Event<NovelRT::Timing::Timestamp> eventWithTimestamp =
-            reinterpret_cast<NovelRT::Utilities::Event<NovelRT::Timing::Timestamp>&>(event);
-        time.tick(eventWithTimestamp);
-        return NRT_SUCCESS;
+        Nrt_setErrIsNullInstanceProvidedInternal();
+        return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
     }
 
-#ifdef __cplusplus
+    auto* time = reinterpret_cast<NovelRT::Timing::StepTimer*>(timer);
+    auto* eventWithTimestamp = reinterpret_cast<NovelRT::Utilities::Event<NovelRT::Timing::Timestamp>*>(event);
+    time->tick(*eventWithTimestamp);
+    return NRT_SUCCESS;
 }
```

Comment:
I do wonder if we should be fixing the casing on the function names here since we're updating the C API but maybe that should be a separate issue.

Reply:
Separate issue, but good callout :)

Reply:
Created #547 for this